src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs

```diff
@@ -46,10 +47,41 @@ public static CompletionItem Create(INamedTypeSymbol typeSymbol, string containi
                  tags: GlyphTags.GetTags(typeSymbol.GetGlyph()),
                  rules: CompletionItemRules.Default,
                  displayTextPrefix: null,
-                 displayTextSuffix: typeSymbol.Arity == 0 ? string.Empty : "<>",
+                 displayTextSuffix: typeSymbol.Arity == 0 ? string.Empty : GetDisplayTextSuffixForGeneric(isCSharp),
                  inlineDescription: containingNamespace);
         }
 
+        public static CompletionItem CreateAttributeNameItem(CompletionItem attributeItem, string attributeName)
+        {
+            Debug.Assert(!attributeItem.Properties.ContainsKey(AttributeFullName));
+
+            var newProperties = attributeItem.Properties.Add(AttributeFullName, attributeItem.DisplayText);
+
+            var sortTextBuilder = PooledStringBuilder.GetInstance();
+            sortTextBuilder.Builder.AppendFormat(SortTextFormat, attributeName, attributeItem.InlineDescription);
+
+            return CompletionItem.Create(
+                 displayText: attributeName,
+                 filterText: attributeName,
+                 sortText: sortTextBuilder.ToStringAndFree(),
+                 properties: newProperties,
+                 tags: attributeItem.Tags,
+                 rules: attributeItem.Rules,
+                 displayTextPrefix: attributeItem.DisplayTextPrefix,
+                 displayTextSuffix: attributeItem.DisplayTextSuffix,
+                 inlineDescription: attributeItem.InlineDescription);
+        }
+
+        public static CompletionItem CreateItemWithGenericDisplaySuffix(CompletionItem item, bool isCSharp)
+        {
+            return item.WithDisplayTextSuffix(GetDisplayTextSuffixForGeneric(isCSharp));
+        }
+
+        private static string GetDisplayTextSuffixForGeneric(bool isCSharp)
+        {
+            return isCSharp ? "<>" : "(Of ...)";
+        }
```

Comment:
Looks weird. We should define an abstract here but implement on a language level.

Reply:
fixed